Exception with latest breathe version

Hi,

we use breathe in our project ([NetworKit](https://github.com/networkit/networkit)) to generate the documentation website. As of today (breathe version v4.29.1) the process fails with an assertion-error.

```
Exception occurred:
  File "/home/runner/work/networkit/networkit/pyenv/lib/python3.8/site-packages/breathe/renderer/sphinxrenderer.py", line 2113, in visit_docparamlist
    assert len(paramNameNodes) <= 1, paramNameNodes
```

Log from Github Actions: https://github.com/networkit/networkit/runs/2417500904#step:4:6624

We install breathe and all other components ([link to environment](https://github.com/networkit/networkit/blob/master/.github/workflows/scripts/documentation.sh)) via pip3. 

Link to sphinx [conf.py](https://github.com/networkit/networkit/blob/master/docs/conf.py).

Is there anything we have to change? Or should we pin the version to an older release?

Thanks in advance!

Damn, I tried to be careful when checking the XML structure. This assertion should be relaxed to a warning that something will look wrong.
To fix it we need to dig into the Doxygen XML and see what is going on for this case.
Is there a way to get to a state were we can just run ``doxygen`` and ``sphinx-build`` for your project? I tried just running ``.github/workflows/scripts/documentation.sh`` but I got
```
CMake Error at CMakeLists.txt:269 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
```

I also encountered the problem during build of https://github.com/nanodbc/nanodbc/ documentation

- Sphinx 3.5.4
- Breathe 4.29.1

```console
sphinx-build -b html -d build/doctrees   . build/html
Running Sphinx v3.5.4
making output directory... done
warning: ignoring unsupported tag `DOXYGEN=' at line 17, file nanodbc.cfg
/home/circleci/project/nanodbc/nanodbc.h:1973: warning: Compound nanodbc::datasource is not documented.
/home/circleci/project/nanodbc/nanodbc.h:164: warning: Member NANODBC_TEXT(s) (macro definition) of file nanodbc.h is not documented.
/home/circleci/project/nanodbc/nanodbc.h:219: warning: Member NANODBC_DEPRECATED (macro definition) of file nanodbc.h is not documented.
/home/circleci/project/nanodbc/nanodbc.h:163: warning: Member string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:173: warning: Member wide_string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:177: warning: Member wide_char_t (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:187: warning: Member null_type (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:368: warning: Member enable_if_string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:371: warning: Member enable_if_character (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:368: warning: Member enable_if_string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:371: warning: Member enable_if_character (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:163: warning: Member string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:173: warning: Member wide_string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:177: warning: Member wide_char_t (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:187: warning: Member null_type (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:368: warning: Member enable_if_string (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:371: warning: Member enable_if_character (typedef) of namespace nanodbc is not documented.
/home/circleci/project/nanodbc/nanodbc.h:294: warning: Member what() const noexcept (function) of class nanodbc::database_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:295: warning: Member native() const noexcept (function) of class nanodbc::database_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:296: warning: Member state() const noexcept (function) of class nanodbc::database_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:272: warning: Member what() const noexcept (function) of class nanodbc::index_range_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:263: warning: Member what() const noexcept (function) of class nanodbc::null_access_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:280: warning: Member programming_error(const std::string &info) (function) of class nanodbc::programming_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:281: warning: Member what() const noexcept (function) of class nanodbc::programming_error is not documented.
/home/circleci/project/nanodbc/nanodbc.h:254: warning: Member what() const noexcept (function) of class nanodbc::type_incompatible_error is not documented.
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 5 source files that are out of date
updating environment: [new config] 5 added, 0 changed, 0 removed
reading sources... [ 20%] api

Exception occurred:
  File "/usr/local/lib/python3.8/site-packages/breathe/renderer/sphinxrenderer.py", line 2113, in visit_docparamlist
    assert len(paramNameNodes) <= 1, paramNameNodes
AssertionError: [<breathe.parser.compound.docParamNameSub object at 0x7f8b11e86d60>, <breathe.parser.compound.docParamNameSub object at 0x7f8b11e86dc0>]
The full traceback has been saved in /tmp/sphinx-err-98xe1y6_.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:62: html] Error 2
```

--- a/breathe/renderer/sphinxrenderer.py
+++ b/breathe/renderer/sphinxrenderer.py
@@ -2105,26 +2105,38 @@ class SphinxRenderer:
         for item in node.parameteritem:
             # TODO: does item.parameternamelist really have more than 1 parametername?
             assert len(item.parameternamelist) <= 1, item.parameternamelist
-            nameNodes = []
+            nameNodes: List[Node] = []
             parameterDirectionNodes = []
             if len(item.parameternamelist) != 0:
                 paramNameNodes = item.parameternamelist[0].parametername
-                # TODO: how many elements can paramNameNodes have?
-                assert len(paramNameNodes) <= 1, paramNameNodes
                 if len(paramNameNodes) != 0:
-                    content = paramNameNodes[0].content_
-                    # this is really a list of MixedContainer objects, i.e., a generic object
-                    # we assume there is either 1 or 2 elements, if there is 2 the first is the
-                    # parameter direction
-                    assert len(content) == 1 or len(content) == 2, content
-                    nameNodes = self.render(content[-1])
-                    if len(content) == 2:
-                        dir = ''.join(n.astext() for n in self.render(content[0])).strip()
-                        assert dir in ('[in]', '[out]', '[inout]'), ">" + dir + "<"
-                        parameterDirectionNodes = [
-                            nodes.strong(dir, dir),
-                            nodes.Text(' ', ' ')
-                        ]
+                    nameNodes = []
+                    for paramName in paramNameNodes:
+                        content = paramName.content_
+                        # this is really a list of MixedContainer objects, i.e., a generic object
+                        # we assume there is either 1 or 2 elements, if there is 2 the first is the
+                        # parameter direction
+                        assert len(content) == 1 or len(content) == 2, content
+                        thisName = self.render(content[-1])
+                        if len(nameNodes) != 0:
+                            if node.kind == 'exception':
+                                msg = "Doxygen \\exception commands with multiple names can not be"
+                                msg += " converted to a single :throws: field in Sphinx."
+                                msg += " Exception '{}' suppresed from output.".format(
+                                    ''.join(n.astext() for n in thisName))
+                                self.state.document.reporter.warning(msg)
+                                continue
+                            nameNodes.append(nodes.Text(", "))
+                        nameNodes.extend(thisName)
+                        if len(content) == 2:
+                            # note, each paramName node seems to have the same direction,
+                            # so just use the last one
+                            dir = ''.join(n.astext() for n in self.render(content[0])).strip()
+                            assert dir in ('[in]', '[out]', '[inout]'), ">" + dir + "<"
+                            parameterDirectionNodes = [
+                                nodes.strong(dir, dir),
+                                nodes.Text(' ', ' ')
+                            ]
 
             name = nodes.field_name(
                 '', nodes.Text(fieldListName[node.kind] + ' '),

